models/SignupForm.php

```diff
@@ -40,10 +40,37 @@ public function signup()
             $user->display_name = $this->username;
             $user->email = $this->email;
             $user->setPassword($this->password);
-            $user->save(false);
+            if (!$user->save(false)) {
+                return null;
+            }
+
+            $this->sendEmailVerificationEmail($user);
+
             return $user;
         }
 
         return null;
     }
+
+    private function sendEmailVerificationEmail(User $user)
```

Comment:
this is duplicated with the code from UserController, should be a single method.